docs/data/date-pickers/base-concepts/ReferenceDateExplicitDateTimePicker.tsx

```diff
@@ -18,7 +19,7 @@ export default function ReferenceDateExplicitDateTimePicker() {
           referenceDate={dayjs('2022-04-17T15:30')}
         />
         <Typography>
-          Stored value: {value == null ? 'null' : value.format()}
+          Stored value: {value == null ? 'null' : (value as Dayjs).format()}
```

Comment:
For me this is the main downside of the new approach

In our doc (and only in our doc) we have several adapter loaded so `PickerValidDate` equals `Moment | DateTime | Dayjs | Date` which means it's harder to manually interact with a date object and we often need to cast.